centreon/src/Core/Domain/Common/GeoCoords.php

```diff
@@ -82,4 +90,24 @@ public static function fromString(string $coords): self
 
         return new self($parts[0], $parts[1]);
     }
+
+    /**
+     * Validate decimals count for a coordinate value
+     *
+     * @param string $value
+     * @param string $type latitude or longitude
+     *
+     * @throws \Assert\AssertionFailedException
+     */
+    private function validateDecimals(string $value, string $type): void
+    {
+        if (str_contains($value, '.')) {
+            $decimalPart = explode('.', $value)[1];
+            Assertion::maxLength(
```

Comment:
This is not an auto truncate, it will throw an exception if there is too many decimals